docs/ossm/ossm2-vs-ossm3.md

```diff
@@ -0,0 +1,103 @@
+
+As `Istio` resources are cluster-scoped, they must have unique names to represent unique meshes within the same cluster. The OpenShift Service Mesh 3 operator uses this unique name to create a resource called `IstioRevision` with a name in the format of `{Istio name}` or `{Istio name}-{Istio version}`. Each instance of `IstioRevision` is responsible for managing a single control plane. It does this using Istio's revision labels of the format `istio.io/rev={IstioRevision name}` to identify pods that are part of their control plane. The name with the version identifier becomes important to support canary-style control plane upgrades (more on this in the upgrades section below).
+
+## Sidecar Injection: New Considerations
+
+OpenShift Service Mesh 2 supported using pod annotations and labels to configure sidecar injection and there was no need to indicate which control plane a workload belonged to.
+
+Sidecar injection in OpenShift Service Mesh 3 works the same way as it does for Istio - with pod or namespace labels used to trigger sidecar injection and it may be necessary to include a label that indicates which control plane the workload belongs to.  Note that Istio has deprecated pod annotations in favor of labels for sidecar injection.
+
+When an `Istio` resource has the name “default”, the label `istio-injection=enabled` may be used.
+
+However, when an Istio resource has a name other than “default” - as required when multiple control plane instances are present and/or a canary-style control plane upgrade is in progress, it is necessary to use a label that indicates which control plane (revision) the workload(s) belong to - namely, `istio.io/rev=<IstioRevision-name>`. These labels may be applied at the workload or namespace level.
+
+## Independently Managed Gateways
+
+In Istio, gateways are used to manage traffic entering (ingress) and exiting (egress) the mesh. While by default, OpenShift Service Mesh 2 deployed and managed an Ingress Gateway and an Egress Gateway with the service mesh control plane, configured in the `ServiceMeshControlPlane` resource, the OpenShift Service Mesh 3 operator will no longer create or manage gateways. 
+
+In OpenShift Service Mesh 3, gateways are created and managed independent of the operator and control plane using gateway injection or Kubernetes Gateway API resources, both of which provide much greater flexibility than was possible with the ServiceMeshControlPlane. This allows the gateways to be customized for a set of applications, deployed and managed with the same lifecycle as their corresponding applications.
+
+This change was made because, as a good practice, gateways are better managed together with their corresponding worklods than with the service mesh control plane. This change also means starting with a gateway configuration that can expand over time to meet the more robust needs of a production environment, which was not possible with the default gateways in OpenShift Service Mesh 2. 
```

Comment:
```suggestion
This change was made because, as a good practice, gateways are better managed together with their corresponding workloads than with the service mesh control plane. This change also means starting with a gateway configuration that can expand over time to meet the more robust needs of a production environment, which was not possible with the default gateways in OpenShift Service Mesh 2. 
```